Cython/Compiler/Tests/TestGrammar.py

```diff
@@ -109,14 +109,16 @@ def test_invalid_number_literals(self):
         for literal in INVALID_UNDERSCORE_LITERALS:
             for expression in ['%s', '1 + %s', '%s + 1', '2 * %s', '%s * 2']:
                 code = 'x = ' + expression % literal
-                try:
+                msg = "Invalid Cython code '%s'" % code
+
+                with self.assertRaises(CompileError, msg=msg) as cm:
                     self.fragment(u'''\
                     # cython: language_level=3
                     ''' + code)
```

Comment:
This `msg = msg` is not supported by 2.7, so the output will be `AssertionError: Exception not raised` rather than `AssertionError: Exception not raised : Invalid Cython code ...` as for other tested versions.
I think this is negligible compared to improving the quality of the code.
@scoder If you also do not think this is especially important, you can resolve this conversation.